Description: Use the SOURCE_DATE_EPOCH variable to set the current year displayed in the javadoc footer
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
+++ b/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
@@ -3044,7 +3044,21 @@
             .map( instant -> instant.atZone( ZoneOffset.UTC ).toLocalDate() )
             .orElseGet( LocalDate::now );
 
-        final String currentYear = Integer.toString( localDate.getYear() );
+        String currentYear = Integer.toString( localDate.getYear() );
+        if ( System.getenv("SOURCE_DATE_EPOCH") != null )
+        {
+            try
+            {
+                java.util.Calendar calendar = java.util.Calendar.getInstance( java.util.TimeZone.getTimeZone("UTC") );
+                calendar.setTimeInMillis(Long.parseLong( System.getenv("SOURCE_DATE_EPOCH") ) * 1000);
+                currentYear = calendar.get( java.util.Calendar.YEAR ) + "";
+            }
+            catch (Exception e)
+            {
+                e.printStackTrace();
+            }
+        }
+
 
         String theBottom = StringUtils.replace( this.bottom, "{currentYear}", currentYear );
 
